model/task_history.go

```diff
@@ -301,8 +303,28 @@ func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string,
 			},
 			{"$sort": bson.D{{Name: task.RevisionOrderNumberKey, Value: -1}}},
 			{"$limit": numCommits},
-			{"$unwind": fmt.Sprintf("$%v", task.TestResultsKey)},
-			{"$group": bson.M{"_id": fmt.Sprintf("$%v.%v", task.TestResultsKey, task.TestResultTestFileKey)}},
+			{"$lookup": bson.M{
```

Comment:
Whether we can do this depends on what calls this. If this is called by any UI pages, it may break in cases with a large number of test results. This is why I rewrote a $lookup against the test results collection as an iteration. Could this be a problem here or in any of the other $lookups?

Reply:
I think that's fine, but I think that means this can't be deployed without EVG-2057 closed. Should we feature branch this change?